internal/featuretests/v3/cluster_test.go

```diff
@@ -656,22 +666,48 @@ func TestUnreferencedService(t *testing.T) {
 		},
 	})
 
-	c.Request(clusterType).Equals(&envoy_discovery_v3.DiscoveryResponse{
+	res := c.Request(clusterType)
+	res.Equals(&envoy_discovery_v3.DiscoveryResponse{
 		Resources: resources(t,
 			cluster("default/kuard/80/da39a3ee5e", "default/kuard", "default_kuard_80"),
 		),
 		TypeUrl: clusterType,
 	})
 
 	// This service which is added should not cause a DAG rebuild
-	rh.OnAdd(fixture.NewService("kuard-notreferenced").
-		WithPorts(v1.ServicePort{Port: 80, TargetPort: intstr.FromInt(8080)}),
-	)
+	s2 := fixture.NewService("kuard-notreferenced").
+		WithPorts(v1.ServicePort{Port: 80, TargetPort: intstr.FromInt(8080)})
+	rh.OnAdd(s2)
 
-	c.Request(clusterType).Equals(&envoy_discovery_v3.DiscoveryResponse{
+	res = c.Request(clusterType)
+	res.Equals(&envoy_discovery_v3.DiscoveryResponse{
+		Resources: resources(t,
+			cluster("default/kuard/80/da39a3ee5e", "default/kuard", "default_kuard_80"),
+		),
+		TypeUrl: clusterType,
+	})
+
+	// verifying that deleting a Service that is not referenced by an HTTPProxy,
+	// does not trigger a rebuild
+	rh.OnDelete(s2)
+	res = c.Request(clusterType)
+	res.Equals(&envoy_discovery_v3.DiscoveryResponse{
+		Resources: resources(t,
+			cluster("default/kuard/80/da39a3ee5e", "default/kuard", "default_kuard_80"),
+		),
+		TypeUrl: clusterType,
+	})
+	assertEqualVersion(t, "2", res)
```

Comment:
Per the failing test, looks like the expected version here is actually "1" - the initial add of `s1` does not trigger a DAG build since it's not _yet_ referenced, then the add of the `default/simple` HTTPProxy _does_ trigger a DAG build, then adding/deleting s2 should _not_ trigger a rebuild, leaving us at a version of "1".